Analysis/include/SyncTupleHTT.h

```diff
@@ -156,3 +158,280 @@ INITIALIZE_TREE(htt_sync, SyncTuple, SYNC_DATA)
+        sync().fatJet_eta = static_cast<float>(fatJet.GetMomentum().Eta());
+        sync().fatJet_phi = static_cast<float>(fatJet.GetMomentum().Phi());
+        sync().fatJet_energy = static_cast<float>(fatJet.GetMomentum().E());
+        sync().fatJet_m_pruned = fatJet->m(ntuple::TupleFatJet::MassType::Pruned);
+        //                sync().fatJet_m_filtered = fatJet->m(ntuple::TupleFatJet::MassType::Filtered);
+        //                sync().fatJet_m_trimmed = fatJet->m(ntuple::TupleFatJet::MassType::Trimmed);
+        sync().fatJet_m_softDrop = fatJet->m(ntuple::TupleFatJet::MassType::SoftDrop);
+        sync().fatJet_n_subjets = static_cast<int>(fatJet->subJets().size());
+        sync().fatJet_n_subjettiness_tau1 = fatJet->n_subjettiness(1);
+        sync().fatJet_n_subjettiness_tau2 = fatJet->n_subjettiness(2);
+        sync().fatJet_n_subjettiness_tau3 = fatJet->n_subjettiness(3);
+    } else {
+        sync().fatJet_pt = default_value;
+        sync().fatJet_eta = default_value;
+        sync().fatJet_phi = default_value;
+        sync().fatJet_energy = default_value;
+        sync().fatJet_m_pruned = default_value;
+        sync().fatJet_m_filtered = default_value;
+        sync().fatJet_m_trimmed = default_value;
+        sync().fatJet_m_softDrop = default_value;
+        sync().fatJet_n_subjets = default_int_value;
+        sync().fatJet_n_subjettiness_tau1 = default_value;
+        sync().fatJet_n_subjettiness_tau2 = default_value;
+        sync().fatJet_n_subjettiness_tau3 = default_value;
+    }
+
+
+    sync().topWeight = event->weight_top_pt;
+    sync().shapeWeight = event->weight_pu * event->weight_bsm_to_sm * event->weight_dy * event->weight_ttbar *
+            event->weight_wjets * event->weight_xs;
+    sync().btagWeight = event->weight_btag;
+
+    sync().lhe_n_b_partons = static_cast<int>(event->lhe_n_b_partons);
+    sync().lhe_n_partons = static_cast<int>(event->lhe_n_partons);
+    sync().lhe_HT = event->lhe_HT;
+
+    sync().genJets_nTotal = event->genJets_nTotal;
+    sync().genJets_nStored = static_cast<unsigned>(event->genJets_p4.size());
+    sync().genJets_nStored_hadronFlavour_b = std::min<unsigned>(2, static_cast<unsigned>(
+                std::count(event->genJets_hadronFlavour.begin(), event->genJets_hadronFlavour.end(), 5)));
+    sync().genJets_nStored_hadronFlavour_c = static_cast<unsigned>(
+                std::count(event->genJets_hadronFlavour.begin(), event->genJets_hadronFlavour.end(), 4));
+    sync().jets_nTotal_hadronFlavour_b = event->jets_nTotal_hadronFlavour_b;
+    sync().jets_nTotal_hadronFlavour_c = event->jets_nTotal_hadronFlavour_c;
+    sync().jets_nSelected_hadronFlavour_b = static_cast<unsigned>(
+                std::count(event->jets_hadronFlavour.begin(), event->jets_hadronFlavour.end(), 5));
+    sync().jets_nSelected_hadronFlavour_c = static_cast<unsigned>(
+                std::count(event->jets_hadronFlavour.begin(), event->jets_hadronFlavour.end(), 4));
```

Comment:
Here and above, for gen jets, please use branches like "jets_nTotal_partonFlavour_b", because genJets vector branches are not available after the skimming.